docs/custom-slashing/dd.template.md

```diff
@@ -0,0 +1,83 @@
+# Custom Slashing Design Document
+
+- Owner: @just-mitch
+- Approvers:
+  - @LHerskind
+  - @Maddiaa0
+- Target DD Approval Date: 2025-05-09
+- Target Project Delivery Date: 2025-05-16
+
+## Executive Summary
+
+The L1 contracts currently only "allow" slashing all validators in an epoch if the epoch is never proven.
+
+We want to provide a mechanism for slashing specific validators for not participating in consensus.
+
+## Requirements
+
+The requirements with filled checkboxes are met by the design below.
+
+- [x] There MUST be ready-made L1 contract(s) that can be used to slash specific validators for not participating in consensus.
+- [x] The Aztec Labs node client software ("the node") MUST automatically slash validators for not participating in consensus.
+- [x] It SHOULD be possible to slash more than one validator at a time.
+- [x] Coordinating the slash SHOULD NOT require any coordination between the validators beyond the existing voting/signalling mechanism; each validator SHOULD be able to inspect L1 and its state to determine:
+  - If it agrees with the slash
+  - How/where to vote/signal on L1
+- [x] Node operators SHOULD be able to configure their node to specify thresholds for what they consider "not participating".
+- [x] The "offence" that triggers the slash MAY be specified on L1.
+- [ ] The amount to be slashed MAY be configurable without deploying a new contract.
+- [ ] The threshold of number of validators (M/N) that need to signal/vote for the CustomSlashFactory payload MAY be configurable without deploying a new contract or a governance action.
+
+## Overview
+
+Rename the existing `SlashFactory` contract to `EpochSlashFactory` to denote that it slashes all validators in an epoch. Make a new `CustomSlashFactory` contract which creates a payload to slash a provided list of validators for an explicit offence. The amounts to be slashed for each offence will be provided in the constructor of the `CustomSlashFactory` contract. There will only be one explicit offence right now: "missing attestations". Creating the payload via the `CustomSlashFactory` will emit an event with the payload address, and the validator addresses.
```

Comment:
Why not nuke the current factory? Slashing the members of a committe is just slashing a list of addresses, which is the same as for missing attestations just with a different "type" 🤷. No need to have an extra one just to have it.